src/ray/core_worker/actor_manager.cc

```diff
@@ -56,49 +58,39 @@ std::pair<std::shared_ptr<const ActorHandle>, Status> ActorManager::GetNamedActo
     const std::string &call_site,
     const rpc::Address &caller_address) {
   ActorID actor_id = GetCachedNamedActorID(GenerateCachedActorName(ray_namespace, name));
-  if (actor_id.IsNil()) {
-    // This call needs to be blocking because we can't return until the actor
-    // handle is created, which requires the response from the RPC. This is
-    // implemented using a promise that's captured in the RPC callback.
-    // There should be no risk of deadlock because we don't hold any
-    // locks during the call and the RPCs run on a separate thread.
-    rpc::ActorTableData actor_table_data;
-    rpc::TaskSpec task_spec;
-    const auto status = gcs_client_->Actors().SyncGetByName(
-        name, ray_namespace, actor_table_data, task_spec);
-    if (status.ok()) {
-      auto actor_handle = std::make_unique<ActorHandle>(actor_table_data, task_spec);
-      actor_id = actor_handle->GetActorID();
-      AddNewActorHandle(std::move(actor_handle),
-                        call_site,
-                        caller_address,
-                        /*is_detached*/ true);
-    } else {
-      // Use a NIL actor ID to signal that the actor wasn't found.
-      RAY_LOG(DEBUG) << "Failed to look up actor with name: " << name;
-      actor_id = ActorID::Nil();
-    }
+  if (!actor_id.IsNil()) {
+    return std::make_pair(GetActorHandle(actor_id), Status::OK());
+  }
 
-    if (status.IsTimedOut()) {
-      std::ostringstream stream;
-      stream << "There was timeout in getting the actor handle, "
-                "probably because the GCS server is dead or under high load .";
-      std::string error_str = stream.str();
-      RAY_LOG(ERROR) << error_str;
-      return std::make_pair(nullptr, Status::TimedOut(error_str));
-    }
+  // This call needs to be blocking because we can't return until the actor
+  // handle is created, which requires the response from the RPC. This is
+  // implemented using a promise that's captured in the RPC callback.
+  // There should be no risk of deadlock because we don't hold any
+  // locks during the call and the RPCs run on a separate thread.
+  rpc::ActorTableData actor_table_data;
+  rpc::TaskSpec task_spec;
+  const auto status = gcs_client_->Actors().SyncGetByName(
+      name, ray_namespace, actor_table_data, task_spec);
+  if (status.ok()) {
+    auto actor_handle = std::make_unique<ActorHandle>(actor_table_data, task_spec);
+    actor_id = actor_handle->GetActorID();
+    AddNewActorHandle(std::move(actor_handle),
+                      call_site,
+                      caller_address,
+                      /*is_detached*/ true);
```

Comment:
Must a named actor be a detached actor? I am not sure why `is_detached = True` here.

Reply:
Oh actually maybe the arg is just named poorly. I think it is more about whether we own the actor or not, will update.